packages/block-editor/src/components/rich-text/format-toolbar-container.js

```diff
@@ -41,7 +41,7 @@ const FormatToolbarContainer = ( { inline, editableContentElement } ) => {
 	// Render regular toolbar.
 	return (
 		<BlockControls group="inline">
-			<FormatToolbar />
+			<FormatToolbar category="content" />
```

Comment:
Note the approach in https://github.com/WordPress/gutenberg/pull/71058 for restricting formatting controls. If your approach is better and we can have one unified mechanism then we'd need to revert that.

Reply:
I think they work alongside each other. This `category="content"` says "we want this component to show when in contentOnly mode" and yours is filtering which format controls should be shown within it.